### PR TITLE
View referral

### PR DESCRIPTION
**Pull request by murilovmachado**
Jira: https://bugs.benetech.org/browse/CHI-374

Primary reviewer: @andvirga 

Screen Shot 2020-12-22 at 15 01 42

### CODE REVIEW

**Review by nickhurlburt — APPROVED** [review]
Looks good, though can we fix this console warning?
```
Warning: Failed prop type: Invalid prop `type` of value `referral` supplied to `TimelineIcon`, expected one of ["facebook","web","voice","sms","whatsapp","E-mail","Website forum","Outreach","Walk-in / In person","Post","Bulletin Board","Other","note"]
```

**Review by andvirga — APPROVED** [review]
Looks great!

File: plugin-hrm-form/src/components/case/ViewReferral.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import * as RoutingActions from '../../states/routing/actions';
+import { CaseContainer } from '../../styles/case';
+import ActionHeader from './ActionHeader';
+import SectionEntry from '../SectionEntry';
+
+type OwnProps = {
+  taskSid: string;
+};
+
+const mapStateToProps = (state, ownProps: OwnProps) => {
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/case/ViewReferral.tsx:19]: Here, you can type the state as `state: RootState` (imported from `src/states/`) and that will infer the types of all of the properties of the returned object! (not only `caseState` but `counselorsHash` and `route` too)

File: plugin-hrm-form/src/components/case/ViewReferral.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  changeRoute: RoutingActions.changeRoute,
+};
+
+type Props = OwnProps & ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps;
+
+const ViewReferral: React.FC<Props> = ({ taskSid, tempInfo, route, changeRoute, counselorsHash }) => {
+  if (!tempInfo || tempInfo.screen !== 'view-referral') return null;
+
+  const { counselor, date, referral } = tempInfo.info;
+  const counselorName = counselorsHash[counselor] || 'Unknown';
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/case/ViewReferral.tsx:38]: Maybe `formatName` helper from `src/utils/formatters.js` is useful here?

**Review by GPaoloni — APPROVED** [review]
Looks awesome! Only conflicts I expect between this PR and customization are in `SectionEntry` and `states/routing/types`, so it will be easy to fix them.

**Pull request merged**